docs/decisions/0002-java-folder-structure.md

```diff
@@ -0,0 +1,106 @@
+           Extensions
+           IntegrationTests
+           InternalUtilities
+           SemanticKernel.Abstractions
+           SemanticKernel.MetaPackage
+           SemanticKernel.UnitTests
+           SemanticKernel
+           Skills
+```
+
+| Folder                         | Description |
+|--------------------------------|-------------|
+| Connectors                     | Parent folder for various Connector implementations e.g., AI or Memory services |
+| Extensions                     | Parent folder for SK extensions e.g., planner implementations |
+| IntegrationTests               | Integration tests |
+| InternalUtilities              | Internal utilities i.e., shared code |
+| SemanticKernel.Abstractions    | SK API definitions |
+| SemanticKernel.MetaPackage     | SK common package collection |
+| SemanticKernel.UnitTests       | Unit tests |
+| SemanticKernel                 | SK implementation |
+| Skills                         | Parent folder for various Skills implementations e.g., MS Graph, GRPC, OpenAI, ... |
+
+Some observations:
+
+* The `src` folder is at the very start of the folder structure, which reduces flexibility
+* The use of the `Skills` term is due to change
+
+```bash
+java
+     api-test
+     samples
+     semantickernel-api
+     semantickernel-bom
+     semantickernel-connectors-parent
+     semantickernel-core-skills
+     semantickernel-core
+     semantickernel-extensions-parent
+```
+
+| Folder                              | Description |
+|-------------------------------------|-------------|
+| `api-test`                          | Integration tests and API usage example |
+| `samples`                           | SK samples |
+| `semantickernel-api`                | SK API definitions |
+| `semantickernel-bom`                | SK Bill Of Materials |
+| `semantickernel-connectors-parent`  | Parent folder for various Connector implementations |
+| `semantickernel-core-skills`        | SK core skills (in .Net these are part of the core implementation) |
+| `semantickernel-core`               | SK implementation |
+| `semantickernel-extensions-parent`  | Parent folder for SK extensions e.g., planner implementation |
+
+Some observations:
+
+* Using lowercase folder name with the `-` delimiter is idiomatic Java
+* The `src` folders are located as close as possible to the source files e.g., `semantickernel-api/src/main/java`, this is idiomatic Java
+* Unit tests are contained together with the implementation
+* The samples are located within the `java` folder and each sample runs standalone
+
+## Decision Outcome
+
+Follow these guidelines:
+
+* The folder names will match those used (or planned for .Net) but in the idiomatic Java folder naming convention
+* Use `bom` instead of `MetaPackage` as the latter is .Net centric
+* Use `api` instead of `Abstractions` as the latter is .Net centric
+* Merge `semantickernel-core-skills` and `semantickernel-core` back into a single module name `semantickernel`
```

Comment:
We have a tentative plan to remove "core skills" from the main SK package in dotnet. List some others, it will be included in the metapackage "Microsoft.SemanticKernel", but if you choose to install "Microsoft.SemanticKernel.Core", it will not be included as this is intended to be just strictly the most barebones parts.

Reply:
FYI I just submitted this PR for dotnet today (removed CoreSkills from SK package, moved code to a new Skills.Core nuget):
https://github.com/microsoft/semantic-kernel/pull/1623

Reply:
@shawncal / @johnoliver I've made a change based on your feedback. The new proposal is:

* Rename `semantickernel-core` to `semantickernel` to be consistent with .Net naming
* Move `semantickernel-core-skills` to a new `plugins` folder and rename to `plugins-core`